arch/x86/crc32_fold_pclmulqdq_tpl.h

```diff
@@ -182,5 +182,9 @@ Z_INTERNAL void CRC32_FOLD(crc32_fold *crc, const uint8_t *src, size_t len, uint
         partial_fold(len, &xmm_crc0, &xmm_crc1, &xmm_crc2, &xmm_crc3, &xmm_crc_part);
     }
 
+    /* Sentinel marker that we use for a comparison later, letting us know if
+     * any residual values have been folded up into this checksum that need to
+     * be finalized. In the C & other implementations, this is the actual checksum. */
+    crc->value = 1;
```

Comment:
Having a comment to explain the reason for why this is set to `1` is a good thing, but for the purpose of reviewing, it is still unclear why the value is exactly `1` and not some "magic constant" like elsewhere.

It is unclear what is `later`... So I would expand it to say exactly which function checks that this value is `1` and not any other value.

Reply:
How's that?

Reply:
Where is the `crc->value` actually being checked?

Reply:
Does it need to be wrapped in `#ifdef GUNZIP`?

Reply:
I don't think so, it's a function only actually called when using the crc32 folding routines, which would be enabled anyway with GUNZIP being defined.  Technically these functions are used in the generic crc32 exported function as well, but the extra assignment is hardly consequential, I think.